wallet-new/docs/wallet-tls.md

```diff
@@ -0,0 +1,146 @@
+<div id="table-of-contents">
+<h2>Table of Contents</h2>
+<div id="text-table-of-contents">
+<ul>
+<li><a href="#sec-1">1. Introduction</a></li>
+<li><a href="#sec-2">2. Abstract</a></li>
+<li><a href="#sec-3">3. Motivation</a></li>
+<li><a href="#sec-4">4. Current design</a>
+<li><a href="#sec-5">5. Proposed design</a>
+<ul>
+<li><a href="#sec-5-1">5.1. Nodejs frontend change</a></li>
+<li><a href="#sec-5-2">5.2. Haskell backend change</a></li>
+<ul>
+</li>
+</ul>
+</div>
+</div>
+
+# Introduction<a id="sec-1" name="sec-1"></a>
+
+# Abstract<a id="sec-2" name="sec-2"></a>
+
+The purpose of this document is to review security model in communication
+ between wallet backend and wallet frontend (Daedalus).
+
+# Motivation<a id="sec-3" name="sec-3"></a>
+
+Communication between wallet frontend (Daedalus) and wallet backend in cardano
+ is done via http server. There are various attacks possible to this kind of 
+environmnet (TODO footnote or further list of possible attack vectors needed).
+
+# Current design<a id="sec-4" name="sec-4"></a>
+
+This summarises current approach in server-client communication.
+
+-   wallet server has https server (based on TLS CA server cert) set up and listening on specific IP:PORT
+    
+-   client is using its on pair of CA certificate in order to verify server cert when sending request to server
+
+Downsides:
+
+-   any other user on the same machine can connect to the wallet backend. If there's no spending password they can also spend
+
+# Proposed design<a id="sec-5" name="sec-5"></a>
+
+From [devops-103](https://iohk.myjetbrains.com/youtrack/issue/DEVOPS-103):
+
+-   to defend against the most obvious of impersonation attacks, we need the wallet's frontend and backend to mutually authenticate to each other
+
+-   the previous scheme, partly based on TLS CA server cert (provided by the backend) and the authentication password (provided by the frontend, CSM-100) was deemed too be unnecessarily hostile to the user (who had to remember Yet Another Password)
+
+-   previous point means we have to make do with a non-interactive mutual authentication scheme
+
+-   PKI provides us with such a scheme – the CA might be used to sign both the backend key/certificate pair (which we already do in DEVOPS-33/CSM-116) and the frontend key/certificate pair
+
+
+The proposal is to implement this scheme:
+
+-   Instead of three artifacts (CA cert, backend key, backend cert), the installer will generate five (+frontend key, +frontend cert) - already implemented
```

Comment:
Do you have at hand the link to the PR which ships this? I guess it's one of those things really useful to see, as a "witness".